content/download-self.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+// preventing reregistration if reinjected by tabs.executeScript for whatever reason, just in case
+if (typeof self.oldCode !== 'string') {
+  chrome.runtime.onConnect.addListener(port => {
+    if (port.name !== 'downloadSelf') return;
+    const read = r => r.status === 200 ? r.text() : Promise.reject(r.status);
```

Comment:
Let's use a real object with a better message:
```js
Promise.reject(new Error(`fetch error: invalid status ${r.status}`))
```

Reply:
Let us not. `Error` is not transferable. Also there's no need to reject with an `Error` as any rejection is an error. This advice you're trying to follow is for the case when there's only one result variable that can take both data and error.

Reply:
> any rejection is an error

`Promise.reject` doesn't wraps the content to an error object. We still have to create an error object manually.

Reply:
Like I said there's no need to use an `Error` object here. It doesn't offer any benefits except unnecessary complications in this case. No need to blindly follow "best practices", it's actually counter-productive and harmful.